source/windows/secure_channel_tls_handler.c

```diff
@@ -1142,9 +1192,95 @@ static int s_do_application_data_decrypt(struct aws_channel_handler *handler) {
             struct aws_channel_slot *slot = handler->slot;
             aws_channel_shutdown(slot->channel, AWS_OP_SUCCESS);
             error = AWS_OP_SUCCESS;
+        }
+        if (status == SEC_I_RENEGOTIATE) {

```

Comment:
Similar to the existing statues, can we get a brief explanation of what this means?

In fact the whole giant block desperately needs explanatory comments.